lib/modules/apostrophe-palette-widgets/public/js/leanUser.js

```diff
@@ -33,109 +33,118 @@ if (window.$) {
     });
   };
 
-  function init ($form, $tag, schema, piece) {
-    return apos.schemas.populate($form, schema, piece, function (err) {
-      if (err) {
-        apos.notify('A problem populating/enhancing the palette field', { type: 'error', dismiss: true });
-        return;
-      }
-
-      listen($form, $tag, schema, piece);
-
-    });
-  };
-
-  // Listen for changes to form elements within the palette tool
-  function listen ($form, $tag, schema, piece) {
-    $form.find(':input:not([data-apos-workflow-field-state-control])').change(function () {
-      var $field = $(this).parents('[data-apos-palette-field]');
-      var fieldValue = $field.find(':input:not([data-apos-workflow-field-state-control])').val();
-      var fieldSchema = _.find(schema, function (field) {
-        if ($field.attr('data-name') === field.name) {
-          return field;
-        }
-      });
-      submit($field, fieldValue, fieldSchema, $tag);
-    });
-  };
-
   // One debounced save function for each field.
   // If we didn't break them out this way then saves
   // for different fields around the same time might
   // never happen.
-  var debouncedSaves = {};
-
-  // Handles prepping the current front end for changes as well as sending
-  // updates to the apostrophe-global piece
-  function submit ($field, fieldValue, fieldSchema, $tag) {
-    apply($field, fieldValue, fieldSchema, $tag);
-    var saveField = debouncedSaves[fieldSchema.name];
-    if (!saveField) {
-      saveField = _.debounce(save, 250, { leading: true, trailing: true });
-      debouncedSaves[fieldSchema.name] = saveField;
-    }
-    save($field, fieldValue, fieldSchema, $tag);
-  };
-
-  // Append new CSS rule to the end of our generated stylesheet. Last rule wins!
-  function apply ($field, fieldValue, fieldSchema, $tag) {
-    var selector = $field.attr('data-apos-palette-selector');
-    var property = $field.attr('data-apos-palette-property');
-    var mediaQuery = $field.attr('data-apos-palette-media-query');
-    var unit = $field.attr('data-apos-palette-unit') || "";
-    var valueTemplate = $field.attr('data-apos-value-template') || false;
-    var value = $field.find(':input:not([data-apos-workflow-field-state-control])').val();
+  apos.paletteWidgets.debouncedSaves = {};
```

Comment:
Since they were no longer in the same block scope this felt better to keep a consistent reference to the obj.

Reply:
see https://github.com/apostrophecms/apostrophe-palette/pull/20/commits/562570c88d4180b6e12e456e7f019b6f0e25b0f8